anta/tests/snmp.py

```diff
@@ -339,3 +340,74 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"The following SNMP error counters are not found or have non-zero error counters:\n{error_counters_not_ok}")
+
+
+class VerifySnmpLogging(AntaTest):
+    """Verifies SNMP logging and SNMP manager(host) details.
+
+    This test performs the following checks for each specified host:
+
+     1. Verifies that the SNMP logging is enabled on the device.
+     2. Verifies SNMP host matches the expected value.
+     3. Ensures that VRF provided matches the expected value.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all of the following conditions are met:
+        - The SNMP logging is enabled on the device.
+        - The SNMP host matches the expected value.
+        - The VRF provided matches the expected value.
+    * Failure: The test will fail if any of the following conditions is met:
+        - The SNMP logging is disabled on the device.
+        - The SNMP host do not matches the expected value.
+        - The VRF provided do not matches the expected value.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.snmp:
+      - VerifySnmpLogging:
+          hosts:
+            - hostname: 192.168.1.100
+              vrf: default
+            - hostname: 192.168.1.103
+              vrf: MGMT
+    ```
+    """
+
+    categories: ClassVar[list[str]] = ["snmp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show snmp", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifySnmpLogging test."""
+
+        hosts: list[SNMPHost]
+        """List of SNMP hosts."""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifySnmpLogging."""
+        self.result.is_success()
+
+        command_output = self.instance_commands[0].json_output.get("logging", {})
+        # If SNMP logging is disabled, test fails.
+        if not command_output.get("loggingEnabled"):
+            self.result.is_failure("SNMP logging is disabled")
+            return
+
+        host_details = command_output.get("hosts")
+
+        for host in self.inputs.hosts:
+            hostname = str(host.hostname)
+            vrf = host.vrf
+            actual_snmp_host = host_details.get(hostname, {})
+            actual_vrf = "default" if (vrf_name := actual_snmp_host.get("vrf")) == "" else vrf_name
+
+            # If SNMP host is not configured on the system, test fails.
```

Comment:
I think `device` is usually more appropriate in networking contexts.
```suggestion
            # If SNMP host is not configured on the device, test fails.
```